vnext/Microsoft.ReactNative/Modules/LogBoxModule.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "LogBoxModule.h"
+
+#include "IReactContext.h"
+#include "ReactHost/ReactInstanceWin.h"
+#include "ReactNativeHost.h"
+#include "Utils/Helpers.h"
+
+#include <UI.Xaml.Controls.Primitives.h>
+
+namespace Microsoft::ReactNative {
+
+void LogBox::Show() noexcept {
+  if (!Mso::React::ReactOptions::UseDeveloperSupport(m_context.Properties().Handle())) {
+    return;
+  }
+
+  m_context.UIDispatcher().Post([weakThis = weak_from_this()] {
+    if (auto strongThis = weakThis.lock()) {
+      strongThis->ShowOnUIThread();
+    }
+  });
+}
+
+void LogBox::Hide() noexcept {
+  m_context.UIDispatcher().Post([weakThis = weak_from_this()] {
+    if (auto strongThis = weakThis.lock()) {
+      strongThis->HideOnUIThread();
+    }
+  });
+}
+
+void LogBox::ShowOnUIThread() noexcept {
+  auto contextImpl = winrt::get_self<React::implementation::ReactContext>(m_context.Handle());
+  auto weakInstance = static_cast<Mso::React::ReactContext *>(&contextImpl->GetInner())->GetInnerInstance();
+  auto instance = weakInstance.GetStrongPtr();
+  if (!instance)
+    return;
+  auto weakHost = instance->GetHost();
+  auto host = weakHost.GetStrongPtr();
+  if (!host)
+    return;
+
+  m_logBoxContent = React::ReactRootView();
+  m_logBoxContent.ComponentName(L"LogBox");
+  m_logBoxContent.ReactNativeHost(winrt::make<winrt::Microsoft::ReactNative::implementation::ReactNativeHost>(*host));
```

Comment:
>make [](start = 41, length = 4)

We must not create a new ReactNativeHost. Instead, we should expose it from the ReactContext and keep ReactNativeHost as a weak ptr inside of the ReactHost.